force-app/main/default/classes/GiftSelector_TEST.cls

```diff
@@ -30,6 +30,58 @@
 @IsTest
 private class GiftSelector_TEST {
 
+    @isTest
+    private static void shouldIncludeGiftScheduleFields() {
+        // Arrange: Mock recurring gifts enabled for the org
+        mockRecurringGiftsEnabled(true);
+        GiftSelector selector = new GiftSelector();
+
+        // Act: Retrieve select fields to be used in query
+        List<String> selectFields = selector.fields();
+
+        // Assert: expected gift schedule fields are included
+        System.assert(selectFields.contains(String.valueOf(DataImport__c.Recurring_Donation_Recurring_Type__c)),
+            'Recurring_Donation_Recurring_Type__c should be in list of select fields');
+        System.assert(selectFields.contains(String.valueOf(DataImport__c.Recurring_Donation_Date_Established__c)),
+            'Recurring_Donation_Date_Established__c should be in list of select fields');
+        System.assert(selectFields.contains(String.valueOf(DataImport__c.Recurring_Donation_Day_of_Month__c)),
+            'Recurring_Donation_Day_of_Month__c should be in list of select fields');
+        System.assert(selectFields.contains(String.valueOf(DataImport__c.Recurring_Donation_Planned_Installments__c)),
+            'Recurring_Donation_Planned_Installments__c should be in list of select fields');
+        System.assert(selectFields.contains(String.valueOf(DataImport__c.Recurring_Donation_Installment_Period__c)),
+            'Recurring_Donation_Installment_Period__c should be in list of select fields');
+        System.assert(selectFields.contains(String.valueOf(DataImport__c.Recurring_Donation_Installment_Frequency__c)),
+            'Recurring_Donation_Installment_Frequency__c should be in list of select fields');
+        System.assert(selectFields.contains(String.valueOf(DataImport__c.Recurring_Donation_Effective_Date__c)),
+            'Recurring_Donation_Effective_Date__c should be in list of select fields');
+    }
+
+    @isTest
+    private static void shouldNotIncludeGiftScheduleFields() {
+        // Arrange: Mock recurring gifts disabled for the org
+        mockRecurringGiftsEnabled(false);
+        GiftSelector selector = new GiftSelector();
```

Comment:
What do you think about creating different selectors that are used for different purposes? It's starting to feel like this selector is pretty configurable, and it might be easier to work with if the different selectors were named in an expressive way and each one had a static list of fields that it used as opposed to querying different fields for different purposes.

Also the Strings in the Assertions should probably be using string.valueof() also.

Reply:
Completely agree with the different selectors! At the very least I think this particular class will need to be updated/renamed accordingly as its responsibilities are reduced.